### ISSUE TITLE
Rustdoc should emit a warning if a `main` function is surrounded by non-items in a doctest

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#140310]
For example:

```rust
//! ```
//! let x = 12;
//!
//! fn main() {
//!     // this code is never run
//! }
//! ```
```

In this case, the `main` function will be wrapped into another `main` function and won't be run. We should warn about it.
